src/backend/marsha/core/api.py

```diff
@@ -678,6 +686,110 @@ def initiate_upload(self, request, pk=None):
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        video__id=user.id,
+                        email=user.token.payload["user"]["email"],
+                    )
+                # no email, user has access to this registration if it's the right combinaison of
+                # lti_user_id and lti_consumer_id
+                if (
+                    user.token.payload["user"].get("id", None) is not None
+                    and user.token.payload.get("context_id", None) is not None
+                ):
+                    playlist = Playlist.objects.get(
+                        lti_id=user.token.payload["context_id"]
+                    )
+                    return LiveRegistration.objects.filter(
+                        id=self.kwargs["pk"],
+                        lti_consumer_site=playlist.consumer_site,
+                        lti_user_id=user.token.payload["user"]["id"],
+                        video__id=user.id,
+                    )
+
+        return LiveRegistration.objects.none()
+
+    def perform_create(self, serializer):
+        """Use data from token before saving."""
+        user = self.request.user
+        if user.token.payload.get("user", None) is not None:
+            email = self.request.data["email"]
+            lti_user_id = self.request.data.get("lti_user_id", None)
+            lti_consumer_site = self.request.data.get("lti_consumer_site", None)
+            # if email is present in token, it's assigned to data, value sent gets ignored
+            if user.token.payload["user"].get("email", None) is not None:
+                email = user.token.payload["user"]["email"]
+            # if user and context informations are present in token it's assign to data
+            if (
+                user.token.payload["user"].get("id", None) is not None
+                and user.token.payload.get("context_id", None) is not None
+            ):
+
+                lti_user_id = user.token.payload["user"]["id"]
+                lti_consumer_site = Playlist.objects.get(
+                    lti_id=user.token.payload["context_id"]
+                ).consumer_site
+            return serializer.save(
+                email=email,
+                lti_consumer_site=lti_consumer_site,
+                lti_user_id=lti_user_id,
+                should_send_reminders=self.request.data["should_send_reminders"],
+                video=Video.objects.get(pk=user.id),
+            )
+
+        return serializer.save()
+
+    def create(self, request, *args, **kwargs):
+        """Register an email for a live. Restrict registration to scheduled video."""
+        user = request.user
+        try:
+            video = Video.objects.get(pk=user.id)
+        except Video.DoesNotExist:
+            return Response(
+                {"reason": "video with id {id} does not exist".format(id=user.id)},
+                status=404,
+            )
```

Comment:
The serializer is aware of this constaint and will throw an error if the video does not exists.